src/Accordion/amp/Accordion.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AccordionSection from './AccordionSection';
+import AccordionSectionTrigger from './AccordionSectionTrigger';
+import AccordionSectionContent from './AccordionSectionContent';
+
+class Accordion extends Component {
+  constructor(props) {
+    super(props);
+    this.groupIndex = 0;
+    this.groupName = this.getGroupName();
+  }
+
+  getGroupName = () => {
+    this.groupIndex += 1;
+    return `__LEAF_UI__accordion-group-${this.groupIndex}`;
+  }
+
+  render() {
+    const { children } = this.props;
+    return (
+      <div>
+        {
+          React.Children.map(children, (child) => child && child.type.name === 'AccordionSection' ? (
```

Comment:
Comparing type.name sadly does not work in production since the name's are minified.
So the way to do this is to compare the instance types directly like so,
`child.type === AccordionSection`

Reply:
Instead, use the reactChildrenByType implementation, as used in Tooltip and Dropdown, it's much cleaner and simpler to understand